plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt

```diff
@@ -88,12 +89,17 @@ import software.aws.toolkits.resources.message
 import software.aws.toolkits.telemetry.CodewhispererCompletionType
 import software.aws.toolkits.telemetry.CodewhispererSuggestionState
 import software.aws.toolkits.telemetry.CodewhispererTriggerType
+import java.util.concurrent.CancellationException
 import java.util.concurrent.TimeUnit
 
 @Service
 class CodeWhispererService(private val cs: CoroutineScope) : Disposable {
     private val codeInsightSettingsFacade = CodeInsightsSettingsFacade()
     private var refreshFailure: Int = 0
+    private val ongoingRequests = mutableMapOf<Int, InvocationContext?>()
```

Comment:
what's the case when InvocationContext being null

Reply:
before the InvocationContext for the specific jobId is added, it can be null, we did some null check to know if it's first request/response.